src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -183,7 +183,7 @@ protected virtual void AddActorToAttributes(ICollection<SamlAttribute> attribute
         /// <returns>'true' if the string has a start element equal <see cref="SamlConstants.Elements.Assertion"/>.</returns>
         public override bool CanReadToken(string securityToken)
         {
-            if (string.IsNullOrWhiteSpace(securityToken) || securityToken.Length > MaximumTokenSizeInBytes)
+            if (string.IsNullOrWhiteSpace(securityToken) || securityToken.Length * 2 > MaximumTokenSizeInBytes)
```

Comment:
Instead of multiplying by the magic number '2', I would recommend using 'sizeof(char)'.